contracts/0.4.24/Lido.sol

```diff
@@ -355,323 +436,301 @@ contract Lido is ILido, StETH, AragonApp {
     }
 
     /**
-    * @notice Set fee rate to `_feeBasisPoints` basis points.
-    * The fees are accrued when:
-    * - oracles report staking results (beacon chain balance increase)
-    * - validators gain execution layer rewards (priority fees and MEV)
-    * @param _feeBasisPoints Fee rate, in basis points
-    */
-    function setFee(uint16 _feeBasisPoints) external {
-        _auth(MANAGE_FEE);
+     * @dev Set max positive rebase allowed per single oracle report
+     * token rebase happens on total supply adjustment,
+     * huge positive rebase can incur oracle report sandwitching.
+     *
+     * stETH balance for the `account` defined as:
+     * balanceOf(account) = shares[account] * totalPooledEther / totalShares = shares[account] * shareRate
+     *
+     * Suppose shareRate changes when oracle reports (see `handleOracleReport`)
+     * which means that token rebase happens:
+     *
+     * preShareRate = preTotalPooledEther() / preTotalShares()
+     * postShareRate = postTotalPooledEther() / postTotalShares()
+     * R = (postShareRate - preShareRate) / preShareRate
+     *
+     * R > 0 corresponds to the relative positive rebase value (i.e., instant APR)
+     *
+     * NB: The value is not set by default (explicit initialization required),
+     * the recommended sane values are from 0.05% to 0.1%.
+     *
+     * @param _maxTokenPositiveRebase max positive token rebase value with 1e9 precision:
+     *   e.g.: 1e6 - 0.1%; 1e9 - 100%
+     * - passing zero value is prohibited
+     * - to allow unlimited rebases, pass max uint256, i.e.: type(uint256).max
+     */
+    function setMaxPositiveTokenRebase(uint256 _maxTokenPositiveRebase) external {
+        _auth(MANAGE_MAX_POSITIVE_TOKEN_REBASE_ROLE);
+        _setMaxPositiveTokenRebase(_maxTokenPositiveRebase);
+    }
 
-        _setBPValue(FEE_POSITION, _feeBasisPoints);
-        emit FeeSet(_feeBasisPoints);
+    struct OracleReportInputData {
```

Comment:
Add comment that we are trying to overcome stack too deep

Reply:
done